### DISCUSSION TITLE
When will be a Windows release?

### DISCUSSION THREAD

**Discussion by Bellisario** [oven-sh/bun#361] [Q&A] [answered]
I'm really excited about Bun, but at the moment I only have a Windows 7 computer (that doesn't even support WSL)...
Will be a Windows release? If yes, when? (I can't wait for it :stuck_out_tongue_winking_eye:)

---
CC: @Jarred-Sumner

**Comment by Cristiano100** [comment]
There will be a windows release for bun, but there is no time estimation. 
As per the [roadmap](https://github.com/Jarred-Sumner/bun/issues/159)

**Comment by kosperera** [comment, marked as answer]
For the moment, the only way to use `bun` is to have WSL. However, the easiest way to do that is to have a development container, like VS Code - Remote Container.

This is all I had to do:

1. Ensure the device supports Virtualization and Hyper-V is enabled.
2. Install Docker CLI or Docker Desktop or Rancher Desktop (Opensource)
3. Install WSL 2 + Ubuntu distro
  > I need this to use Linux-based containers, or else I'll end up only with windows-based containers.
  > https://docs.microsoft.com/en-us/windows/wsl/setup/environment
4. Install VS Code + Docker and Remote - Container extensions
  > I already had these installed since I do most of the .NET and JS stuff on dev containers
5. Fork or use https://github.com/alertbox/try-bun to create a copy of the quick-starter project
  > This is a template repo so I didn't have to Fork it. I just used the `Use this template` feature in GitHub.
6. Clone the newly copied/forked repo
  ```zsh
  # example
  $ gh repo clone <repo-name>
  $ # Or
  $ # git clone https://github.com/<username>/<repo-name>
  ```
7. Open the code in VS Code
  > VS Code automatically recognizes the dev container and prompts you to Reopen in Container.
8. Once loaded, spin up a New Terminal in VS Code
9. Type `$ bun run test-project/server.js`
  > Spin up my favorite browser to visit http://localhost:3000

**Reply by Zamiell** [reply]
Note that WSL2 has a lot of performance problems on Windows. The speedups that you get from Bun may be vastly outweighed by the super slow file-system calls (if your WSL has to access files from your "real" Windows drives).

**Reply by kosperera** [reply]
This worked for me on Windows via WSL2 using Dev containers, the latest versions of both. See [@alertbox/dcosify-footer](https://github.com/alertbox/docsify-footer/)
